src/jitlayers.cpp

```diff
@@ -497,7 +503,36 @@ static void addPassesForOptLevel(legacy::PassManager &PM, TargetMachine &TM, raw
 
 CompilerResultT JuliaOJIT::CompilerT::operator()(Module &M)
 {
+    uint64_t start_time = 0;
+    if (dump_llvm_opt_stream != NULL) {
+        // Print LLVM function statistics _before_ optimization
+        // Print all the information about this invocation as a YAML object
+        jl_printf(dump_llvm_opt_stream, "- \n");
+        // We print the name and some statistics for each function in the module, both
+        // before optimization and again afterwards.
+        jl_printf(dump_llvm_opt_stream, "  before: \n");
+        for (auto &F : M.functions()) {
+            if (F.getInstructionCount() == 0 || F.getName().str().rfind("jfptr_", 0) == 0) {
+                continue;
+            }
+            // Count number of Basic Blocks
+            int bbs = 0;
+            for (auto &B : F.getBasicBlockList()) {
+                std::ignore = B;
+                ++bbs;
+            }
+
+            // Each function is printed as a YAML object with several attributes
+            jl_printf(dump_llvm_opt_stream, "    %s:\n", F.getName().str().c_str());
```

Comment:
```suggestion
            jl_printf(dump_llvm_opt_stream, "    \"%s\":\n", F.getName().str().c_str());
```
(Is this necessary? I'm not well-versed in YAML)

Reply:
Can't hurt, I agree.

I think I left it off because in previous commits I was using CSVs and i didn't know what would happen, and it occurred to me that the c function names can't have any whitespace anyway..

BUT still, i agree it seems nicer to include. Thanks. :)